### PR TITLE
Pretty logs

### PR DESCRIPTION
**Pull request by lemonsqueeze**
[ work-in-progress, don't merge ]

Trying to make logs more readable / useful.
Normal output is more quiet and shows timestamps, --debug shows line numbers and all the extra stuff.

Right now it might looks like this:

```
Apr 15 20:46:39   #  Accepting challenge from stevl (3k)  [19x19]  id = 12478689
Apr 15 20:46:40   [Game 12478689]                 gamedata          [19x19]  W              stevl
Apr 15 20:46:42   [Game 12478689]     Move 1      Got     q16
Apr 15 20:46:42   [Game 12478689]     Move 1      Starting new bot process
Apr 15 20:46:42   [Game 12478689]     Move 1      State loading for new bot
Apr 15 20:46:42   [Game 12478689]     Move 1      genmove w
Apr 15 20:46:47   [Game 12478689]     Move 1      Playing d4
Apr 15 20:46:50   [Game 12478689]     Move 3      Got     q4
Apr 15 20:46:50   [Game 12478689]     Move 3      genmove w
Apr 15 20:46:59   [Game 12478689]     Move 3      Playing d16
Apr 15 20:47:01   [Game 12478689]     Move 5      Got     c6
Apr 15 20:47:01   [Game 12478689]     Move 5      genmove w
Apr 15 20:47:05   [Game 12478689]     Move 5      Playing f3
Apr 15 20:47:08   [Game 12478689]     Move 7      Got     b4
Apr 15 20:47:08   [Game 12478689]     Move 7      genmove w
Apr 15 20:47:11   [Game 12478689]     Move 7      Playing c3
Apr 15 20:47:13   [Game 12478689]     Move 9      Got     b3
Apr 15 20:47:13   [Game 12478689]     Move 9      genmove w
```

Also shows handicap info for handicap games.

Feedback & ideas welcome.

### CODE REVIEW

**Comment by roy7** [comment]
Looks like this needs updated to the new /devel. :)

**Comment by lemonsqueeze** [comment]
Just rebased on new devel, should be better now.

File: gtp2ogs.js [hunk 1]
```diff
@@ -1154,6 +1195,10 @@ class Connection {
                 this.processMove(gamedata);
             } */
 
+	    // Don't connect to old finished games.
```
Comment (roy7) [inline gtp2ogs.js]: I'd been meaning to eventually change this myself. :D Since it fills my logs with connect/disconnect on already finished games.

**Comment by lemonsqueeze** [comment]
Nothing left on my todo list, i'm happy with this =)

One thing i was wondering about though:
At one point i tried to print player rank on /gamedata, but it didn't make sense. I left a comment:

```javascript
        // XXX doesn't work, getting garbage ranks here ...
        // let rank = rank2str(player.rank);
```
The ranks in challenge notification work, but not the ones in game.state. Don't know what's going on here...

**Comment by roy7** [comment]
I merged in this but I seem to get this sort of error on every new bot starting up btw.

> May 05 04:48:32   gtp2ogs.js:353                 [undefined] Starting  /home/ubuntu/leela-zero/src -r 10 -w /home/roy/code/leela-zero/src//home/ubuntu/leela-ze
> ro/src/62b5417b64c46976795d10a6741801f15f857e5029681a42d02c9852097df4b9 --gtp --lagbuffer 300
> May 05 04:48:32 trace gtp2ogs.js:267                 ERROR: Uncaught exception
> May 05 04:48:32 !!!!! gtp2ogs.js:268                 ERROR: TypeError: Cannot read property 'on' of undefined
>

**Comment by roy7** [comment]
I had two gtp2ogs running at same time, oops, maybe that was why.

**Comment by lemonsqueeze** [comment]
Oh ok, good to know. Never ran into this one before =)

**Pull request merged**